src/components/plots/piePlot.ts

```diff
@@ -67,6 +91,102 @@ export module Plots {
+    public sectorValue(sectorValue?: number | _Accessor | D, sectorValueScale?: Scale<D, number>): any {
+      if (sectorValue == null) {
+        return this._key2DataBindings.get(Pie._SECTOR_VALUE_KEY);
+      }
+      this._replaceDataScaleBinding(this.sectorValue().scale, sectorValueScale);
+      this._key2DataBindings.set(Pie._SECTOR_VALUE_KEY, { accessor: d3.functor(sectorValue), scale: sectorValueScale });
+      this._updateDataScaleExtents(Pie._SECTOR_VALUE_KEY);
+      this._render();
+      return this;
+    }
+
+    public innerRadius(): AccessorScaleBinding<D, number>;
+    public innerRadius(innerRadius: number | _Accessor): Plots.Pie<D>;
+    public innerRadius(innerRadius: D | _Accessor, innerRadiusScale: Scale<D, number>): Plots.Pie<D>;
+    public innerRadius(innerRadius?: number | _Accessor | D, innerRadiusScale?: Scale<D, number>): any {
+      if (innerRadius == null) {
+        return this._key2DataBindings.get(Pie._INNER_RADIUS_KEY);
+      }
+      this._replaceDataScaleBinding(this.innerRadius().scale, innerRadiusScale);
+      this._key2DataBindings.set(Pie._INNER_RADIUS_KEY, { accessor: d3.functor(innerRadius), scale: innerRadiusScale });
+      this._updateDataScaleExtents(Pie._INNER_RADIUS_KEY);
+      this._render();
+      return this;
+    }
+
+    public outerRadius(): AccessorScaleBinding<D, number>;
+    public outerRadius(outerRadius: number | _Accessor): Plots.Pie<D>;
+    public outerRadius(outerRadius: D | _Accessor, outerRadiusScale: Scale<D, number>): Plots.Pie<D>;
+    public outerRadius(outerRadius?: number | _Accessor | D, outerRadiusScale?: Scale<D, number>): any {
+      if (outerRadius == null) {
+        return this._key2DataBindings.get(Pie._OUTER_RADIUS_KEY);
+      }
+      this._replaceDataScaleBinding(this.outerRadius().scale, outerRadiusScale);
+      this._key2DataBindings.set(Pie._OUTER_RADIUS_KEY, { accessor: d3.functor(outerRadius), scale: outerRadiusScale });
+      this._updateDataScaleExtents(Pie._OUTER_RADIUS_KEY);
+      this._render();
+      return this;
+    }
+
+    protected _updateExtents() {
+      super._updateExtents();
+      this._key2DataExtents.keys().forEach((dataAttr) => this._updateDataScaleExtents(dataAttr));
+    }
+
+    private _dataExtentsForScale<D>(scale: Scale<D, any>): D[][] {
+      if (!this._isAnchored) {
+        return [];
+      }
+      var allSetsOfExtents: D[][][] = [];
+      this._key2DataBindings.keys().forEach((dataAttr: string) => {
+        if (this._key2DataBindings.get(dataAttr).scale === scale) {
+          var extents = this._key2DataExtents.get(dataAttr);
+          if (extents != null) {
+            allSetsOfExtents.push(extents);
+          }
+        }
+      });
+      return d3.merge(allSetsOfExtents);
+    }
+
+    private _updateDataScaleExtents(dataAttr: string) {
+      this._key2DataExtents.set(dataAttr, this._datasetExtents(dataAttr));
+      var dataScale = this._key2DataBindings.get(dataAttr).scale;
```

Comment:
This line needs to combine with the below `_datasetExtents()` method to form `_updateExtentsForProperty()`, the parallel to `_updateExtentsForAttr()`.